server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -72,6 +73,8 @@
             (value) -> Translog.Durability.valueOf(value.toUpperCase(Locale.ROOT)), Property.Dynamic, Property.IndexScope);
     public static final Setting<Boolean> INDEX_WARMER_ENABLED_SETTING =
         Setting.boolSetting("index.warmer.enabled", true, Property.Dynamic, Property.IndexScope);
+    public static final Setting<Integer> INDEX_MAX_DOC_ID_LENGTH_SETTING =
+        Setting.intSetting("index.max_doc_id_length", 512, 1, Property.Dynamic, Property.IndexScope);
```

Comment:
Rather than making this setting `Dynamic`, I think we should make it `Final` at least for now. As mentioned earlier, this could enable us to optimize for the default id length case in the future. Is there significant use cases you have, where dynamically updating this is necessary?